flink-container/src/main/java/org/apache/flink/container/entrypoint/StandaloneJobClusterConfigurationParserFactory.java

```diff
@@ -49,7 +49,9 @@
 		.required(true)
```

Comment:
Wouldn't it be convenient if we used the manifest as a default fallback? I.e. work just like the normal job-submission; check manifest by default and optionally allow specifying the main-class.

Reply:
How would you specify this? I would eventually be all in for something like:
```
StandaloneJobClusterEntryPoint --job-jar XYZ.jar <mainArgs>
StandaloneJobClusterEntryPoint [--job-classname OptionalClassName] XYZ.jar <mainArgs>
```
But as noted in the PR description I find it ultimately confusing since the JAR  is expected to be on the class path although the arguments give the impression that you can specify any JAR. That's why I suggest the `from-jar:` workaround.

I would be happy to change this in the future if we distribute JARs with this entry point.

Reply:
When I wrote my suggestion I forgot that we don't pass a jar by default like for normal submissions 🤦‍♂️ 

The `from-jar:` workaround seems rather unusual to me. It also makes the `jobClassName` field in the `StandaloneJobClusterConfiguration` ambiguous; it may be a class name or the path to a file with a custom prefix, which are entirely different things.
I would be in favor of a separate argument; the added overhead for ensuring that one of them is set is rather low and I don't want to start adding workarounds just because the underlying library doesn't support case X out-of-the-box. This option could be named `--manifest-jar` or `--from-jar` without a shorthand (to retain the `from-jar` principle being visible).

Reply:
I debated this myself. I still find this workaround preferable (easy to remove, no additional argument, no cumbersome argument checking).

If we go with an argument, I would make it super explicit and name it something like `--job-classname-from-jar` and require users to have either `--job-classname` or `--job-classname-from-jar` (but not both).

What do you think?

Reply:
We'll probably need a tie-breaker here.

IMO,
* the `from-jar` approach isn't significantly harder to remove that a dedicated option; I'd argue it would make a removal cleaner since we wouldn't be removing a special case from an existing option.
* no additional argument is not a benefit in itself
* the argument checking is pretty much just `commandLine.hasOption(op1.getOpt()) ^ commandLine.hasOption(op2.getOpt())`, with a `FlinkParseException` being thrown if the result is false. (You might argue that in order to do so we have to modify the exception signature of `StandaloneJobClusterConfigurationParserFactory#createResult`, but we should do so anyway as we are actually still parsing the rest port and job ID)

In any case we should split the `jobClassName` field to not have ambiguous semantics; this could be modeled as an `Either<String, File>`; the `from-jar` scheme is just a CLI construct to distinguish paths from class names; the `ClassPathJobGraphRetriever ` shouldn't be aware of it.

Reply:
I'm fine with the other approach to here. I was just asking about the naming. 🙈